web/src/containers/ActivityDetailContractorResources.js

```diff
@@ -41,22 +50,36 @@ class ContractorExpenses extends Component {
     updateActivity(activity.key, updates, true);
   };
 
-  handleDrop = index => files => {
+  handleFileUpload = index => files => {
+    if (!files.length) return;
+
     const { activity, updateActivity } = this.props;
+    const { docType } = this.state;
+
+    // only do one file at a time
+    const { name, preview, size, type } = files[0];
+    const newFile = { name, preview, size, type, category: docType };
     const existingFiles = activity.contractorResources[index].files || [];
-    const newFiles = files.map(({ name, preview, size, type }) => ({
-      name,
-      preview,
-      size,
-      type
-    }));
-
-    const updates = { [index]: { files: [...existingFiles, ...newFiles] } };
+
+    const updates = { [index]: { files: [...existingFiles, newFile] } };
+    updateActivity(activity.key, { contractorResources: updates });
+
+    // reset document category if necessary
+    if (docType !== DOC_TYPES[0]) this.setState({ docType: DOC_TYPES[0] });
+  };
+
+  handleFileDelete = (cIdx, fIdx) => () => {
+    const { activity, updateActivity } = this.props;
+    const { files } = activity.contractorResources[cIdx];
+    const updatedFiles = files.filter((_, i) => i !== fIdx);
+
+    const updates = { [cIdx]: { files: updatedFiles } };
```

Comment:
Out of curiosity, is there a reason to do the filtering here instead of the reducer?  Similarly with adding a file.  In the other places we add/remove list items, we do it in the reducer.

Reply:
mainly just so we didn't need to add additional actions and reducer code, though you're right that it's now not quite consistent; i guess i just like utilizing the versatile update action (and `updeep`)